### PR TITLE
Correct MAVLINK_MSG_ID_SET_ATTITUDE_TARGET anglular rate input frame

### PR DESCRIPTION
**Pull request by lthall**
This PR fixes a problem where the angular rate inputs to MAVLINK_MSG_ID_SET_ATTITUDE_TARGET were not in the body frame if both attitude and rate are specified. It also checks to make sure each of the three rotation rate axis are specified.

This fixes the bug where when only rates are specified they are in body frame but when using quaternion attitude and rates the rates were in earth frame.

The message clearly specifies the rates are in body frame.

### CODE REVIEW

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.cpp [hunk 1]
```diff
@@ -149,15 +149,15 @@ void AC_AttitudeControl_Multi_6DoF::input_angle_step_bf_roll_pitch_yaw(float rol
 // Command a Quaternion attitude with feedforward and smoothing
 // attitude_desired_quat: is updated on each time_step (_dt) by the integral of the angular velocity
 // not used anywhere in current code, panic in SITL so this implementation is not overlooked
-void AC_AttitudeControl_Multi_6DoF::input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_target) {
+void AC_AttitudeControl_Multi_6DoF::input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_body) {
```
Comment (tridge) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.cpp:152]: should be:
 const Vector3f &ang_vel_body

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.h [hunk 2]
```diff
@@ -19,9 +19,9 @@ class AC_AttitudeControl_Multi_6DoF : public AC_AttitudeControl_Multi {
     }
 
     // Command a Quaternion attitude with feedforward and smoothing
-    // attitude_desired_quat: is updated on each time_step (_dt) by the integral of the angular velocity
+    // attitude_desired_quat: is updated on each time_step (_dt) by the integral of the body frame angular velocity
     // not used anywhere in current code, panic so this implementaiton is not overlooked
-    void input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_target) override;
+    void input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_body) override;
```
Comment (tridge) [inline libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.h:24]: should be:
  const Vector3f &ang_vel_body

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h [hunk 3]
```diff
@@ -159,8 +159,8 @@ class AC_AttitudeControl {
     void inertial_frame_reset();
 
     // Command a Quaternion attitude with feedforward and smoothing
-    // attitude_desired_quat: is updated on each time_step (_dt) by the integral of the angular velocity
-    virtual void input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_target);
+    // attitude_desired_quat: is updated on each time_step (_dt) by the integral of the body frame angular velocity
+    virtual void input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_body);
```
Comment (tridge) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.h:163]: should be:
 const Vector3f &ang_vel_body

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp [hunk 4]
```diff
@@ -227,15 +227,16 @@ void AC_AttitudeControl::reset_rate_controller_I_terms_smoothly()
 //    trust vector drops below 2*AC_ATTITUDE_THRUST_ERROR_ANGLE. At this point the heading is also corrected.
 
 // Command a Quaternion attitude with feedforward and smoothing
-// attitude_desired_quat: is updated on each time_step by the integral of the angular velocity
-void AC_AttitudeControl::input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_target)
+// attitude_desired_quat: is updated on each time_step by the integral of the body frame angular velocity
+void AC_AttitudeControl::input_quaternion(Quaternion& attitude_desired_quat, Vector3f ang_vel_body)
```
Comment (tridge) [inline libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:231]: should be:
 const Vector3f &ang_vel_body

Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:231]: Done

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.cpp (hunk 1 above)
Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.cpp:152]: Done

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.h (hunk 2 above)
Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.h:24]: Done

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h (hunk 3 above)
Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.h:163]: Done

**Comment by lthall** [comment]
I would love to get a quick explanation of the change to const Vector3f &ang_vel_body

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.cpp (hunk 1 above)
Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.cpp:151]: Changed it and realised that the input quaternion modifies the parameter on the stack. Happy to fix this on all the code I am responsible for in a single PR.

File: libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.h (hunk 2 above)
Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl_Multi_6DoF.h:23]: Changed it and realised that the input quaternion modifies the parameter on the stack. Happy to fix this on all the code I am responsible for in a single PR.

File: libraries/AC_AttitudeControl/AC_AttitudeControl.h (hunk 3 above)
Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.h:164]: Changed it and realised that the input quaternion modifies the parameter on the stack. Happy to fix this on all the code I am responsible for in a single PR.

File: libraries/AC_AttitudeControl/AC_AttitudeControl.cpp (hunk 4 above)
Comment (lthall) [reply libraries/AC_AttitudeControl/AC_AttitudeControl.cpp:231]: Changed it and realised that the input quaternion modifies the parameter on the stack. Happy to fix this on all the code I am responsible for in a single PR.

**Review by tridge — APPROVED** [review]

**Review by rmackay9 — APPROVED** [review]

**Pull request merged**